src/customer/strategies/masterpass-customer-strategy.ts

```diff
@@ -0,0 +1,133 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotImplementedError,
+    NotInitializedError,
+    NotInitializedErrorType
+} from '../../common/error/errors';
+import { PaymentMethod, PaymentMethodActionCreator } from '../../payment';
+import { MasterpassScriptLoader } from '../../payment/strategies/masterpass';
+import { RemoteCheckoutActionCreator } from '../../remote-checkout';
+import CustomerCredentials from '../customer-credentials';
+import { CustomerInitializeOptions, CustomerRequestOptions } from '../customer-request-options';
+
+import CustomerStrategy from './customer-strategy';
+
+export default class MasterpassCustomerStrategy extends CustomerStrategy {
+    private _buttonId?: string;
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { masterpass: masterpassOptions, methodId } = options;
+
+        if (!masterpassOptions || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                if (!this._paymentMethod || !this._paymentMethod.initializationData.checkoutId) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                }
+
+                const cart = state.cart.getCart();
+                if (!cart) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                const { container, buttonId } = masterpassOptions;
+                this._buttonId = (buttonId ? buttonId : 'mpbutton');
+
+                const payload = {
+                    checkoutId: this._paymentMethod.initializationData.checkoutId,
+                    allowedCardTypes: this._paymentMethod.initializationData.allowedCardTypes,
+                    amount: cart.cartAmount.toFixed(2).toString(),
```

Comment:
Is it okay to assume 2 decimal places irrespective of currency?

Reply:
Well, maybe not, should I just leave `.toString()`?